### PR TITLE
Async job 2

### PR DESCRIPTION
**Pull request by Joannitka-pl**
- [x] Added async job for cancelling reservation
- [x] Added tests for async job

### CODE REVIEW

File: app/concepts/reservations/use_cases/create_online.rb [hunk 1]
```diff
@@ -7,4 +7,4 @@ def call(params:)
       end
     end
   end
-end
+end
```
Comment (tobiaszwaszak) [inline app/concepts/reservations/use_cases/create_online.rb:10]: this is something what you shouldn't commit ;)

File: app/concepts/reservations/use_cases/create_reservation_with_tickets.rb [hunk 2]
```diff
@@ -10,12 +10,21 @@ def initialize(repository: Reservations::Repository.new)
       def call(params:)
         Reservation.transaction do
           reservation_params = params.except(:sort, :price)
-          # reservation_params = (params[:seat]).split(',')
           ticket_params = params.except(:status, :client_id, :ticket_desk_id, :seance_id, :seat)
```
Comment (tobiaszwaszak) [inline app/concepts/reservations/use_cases/create_reservation_with_tickets.rb:13]: maybe try to set what params you want to use in `reservation_params` and `ticket_params` 
it's more clear, and easier to maintain if you add more params in future

File: spec/jobs/cancel_reservation_job_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  describe "#perform_later" do
+    let(:destroy_use_case) { instance_double(Reservations::UseCases::Destroy, call: true) }
+
+    it "enqueues reservation timeout process" do
+      ActiveJob::Base.queue_adapter = :test
+      expect { CancelReservationJob.perform_later('reservation')
+             }.to have_enqueued_job
+    end
+
+    it "calls destroy usecase" do
```
Comment (tobiaszwaszak) [inline spec/jobs/cancel_reservation_job_spec.rb:14]: is it works?

**Review by baniak66 — APPROVED** [review]
LGTM 👏

**Comment by tobiaszwaszak** [comment]
check this PR with linter ;)

File: app/concepts/reservations/use_cases/create_reservation_with_tickets.rb [hunk 4]
```diff
@@ -10,12 +10,21 @@ def initialize(repository: Reservations::Repository.new)
...
           @reservation = @repository.create!(reservation_params)
           Tickets::UseCases::Create.new(reservation: @reservation, ticket_params: ticket_params).call
+
+          CancelReservationJob.set(wait_until: seance_expiry_time).perform_later(@reservation.id) unless @reservation.status == "paid"
```
Comment (kodowicz) [inline app/concepts/reservations/use_cases/create_reservation_with_tickets.rb:17]: You have two classes (`CreateOnline` and `CreateOffline`) that inherit from `CreateReservationWithTickets` and both do not implement any extra code. This is not a good approach to separate two classes if they do the same thing. I would consider setting `CancelReservationJob` only into `CreateOnline`. If you assume the seats can be reserved at the cinema and still not be paid automatically I wouldn't split this class into `CreateOnline` and `CreateOffline` until they do not differ.

File: app/concepts/reservations/use_cases/create_online.rb (hunk 1 above)
Comment (Joannitka-pl) [reply app/concepts/reservations/use_cases/create_online.rb:10]: Oh, right. So do you mean doing such changes right through to Main?

File: spec/jobs/cancel_reservation_job_spec.rb (hunk 3 above)
Comment (Joannitka-pl) [reply spec/jobs/cancel_reservation_job_spec.rb:14]: Well, the tests do pass. 😊

File: app/concepts/reservations/use_cases/create_online.rb (hunk 1 above)
Comment (tobiaszwaszak) [reply app/concepts/reservations/use_cases/create_online.rb:10]: you should ignore this kind of change

**Pull request merged**